pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotConnection.java

```diff
@@ -67,15 +74,24 @@ public class PinotConnection extends AbstractBaseConnection {
     }
     _session = new org.apache.pinot.client.Connection(properties, brokers, transport);
 
-    _enableNullHandling = Boolean.parseBoolean(properties.getProperty(QueryOptionKey.ENABLE_NULL_HANDLING));
+    for (String possibleQueryOption: POSSIBLE_QUERY_OPTIONS) {
```

Comment:
Maybe directly get a map of query option from `properties`, and you can use it for any future supported query options as well.

For backward compatibility, you can set `enableNullHandling` if configured as `properties.getProperty(QueryOptionKey.ENABLE_NULL_HANDLING)`

Reply:
With this implementation, to support new query options, an entry per option is added to the POSSIBLE_QUERY_OPTIONS array. All other code remains the same. It also keeps the filtering of query options to the connection creation instead of having to do it per statement execution. It also keeps from having to hash into the options to get the enable flag per statement execution.

I'm not sure how backward compatibility comes into play.